pkg/lint/linter/config.go

```diff
@@ -150,6 +151,12 @@ func (lc *Config) WithNoopFallback(cfg *config.Config) *Config {
 	return lc
 }
 
+func IsGoLowerThan(limit string) func(cfg *config.Config) bool {
```

Comment:
It's here because the signature of this function is only for `WithNoopFallback`.
`IsGoGreaterThanOrEqual` is "neutral" and can be used everywhere.

Reply:
@ldez, imho, inconsistent logic, because `IsGoLowerThan` you made in generic way, not `IsGoLowerThan122`

let's do

```go
func (lc *Config) WithNoopFallback(cfg *config.Config, cond func(cfg *config.Config) (bool, string)) *Config {
    if cfg != nil && cond(cfg) {
		lc.Linter = &Noop{
			name: lc.Linter.Name(),
			desc: lc.Linter.Desc(),
			reason: reason,
    // ...
}

func IsGoLowerThan(limit string) func(cfg *config.Config) (bool, string) {
    return func(cfg *config.Config) bool {
                reason := "This linter is disabled because the Go version of your project is lower than Go " + limit,
		return cfg != nil && !config.IsGoGreaterThanOrEqual(cfg.Run.Go, limit), reason
	}
}
```

or something like that

```go
type disabler interface {
    NeedToDisable(cfg *config.Config) bool   // Apply, Check, etc.
    Reason() string
}

func (lc *Config) WithNoopFallback(cfg *config.Config, d disabler)) *Config {
    if cfg != nil && d.NeedToDisable(cfg) {
		lc.Linter = &Noop{
			name: lc.Linter.Name(),
			desc: lc.Linter.Desc(),
			reason: d.Reason(),
    // ...
}

type tooLowerGo string

func (limit tooLowerGo) NeedToDisable(cfg *config.Config) bool {
    return cfg != nil && !config.IsGoGreaterThanOrEqual(cfg.Run.Go, limit)
}

func (limit tooLowerGo) Reason() string {
    return "This linter is disabled because the Go version of your project is lower than Go " + limit
}

...
WithNoopFallback(m.cfg, tooLowerGo("1.22")),
```

?

Reply:
I will rewrite this part with more deep changes, which I think everybody will agree with.

Reply:
I was not able to do what I wanted because of circular dependencies.
So I created something based on your first suggestion.